pandas/tests/reshape/concat/test_empty.py

```diff
@@ -202,12 +202,15 @@ def test_concat_empty_series_dtypes_sparse(self):
         expected = pd.SparseDtype("object")
         assert result.dtype == expected
 
-    def test_concat_empty_df_object_dtype(self):
+    @pytest.mark.parametrize("dtype", ["int64", "Int64"])
+    def test_concat_empty_df_object_dtype(self, dtype):
         # GH 9149
         df_1 = DataFrame({"Row": [0, 1, 1], "EmptyCol": np.nan, "NumberCol": [1, 2, 3]})
+        df_1["Row"] = df_1["Row"].astype(dtype)
         df_2 = DataFrame(columns=df_1.columns)
         result = pd.concat([df_1, df_2], axis=0)
-        expected = df_1.astype(object)
+        expected = df_1.copy()
+        expected["EmptyCol"] = expected["EmptyCol"].astype(object)  # TODO: why?
```

Comment:
any thoughts on if/why this is the desired behavior?

Reply:
the empty you are concatting with is object dtype

but i think that we should be removing empties during initial stages of concat- so this is a bit sketch - eg i think you should end up with df_1 dtypes but maybe not reading correctly

Reply:
yeah i wouldn't expect this to upcast

Reply:
ill take a look at what it would take to change this, see if it makes sense to do in this PR or separately

Reply:
mm the quick fix i tried for this ended up breaking a bunch of other tests.  let's address the next pass